numpy/core/src/multiarray/ctors.c

```diff
@@ -1862,12 +1869,20 @@ PyArray_CheckFromAny(PyObject *op, PyArray_Descr *descr, int min_depth,
     if (obj == NULL) {
         return NULL;
     }
+
     if ((requires & NPY_ARRAY_ELEMENTSTRIDES) &&
         !PyArray_ElementStrides(obj)) {
-        PyObject *ret;
-        ret = PyArray_NewCopy((PyArrayObject *)obj, NPY_ANYORDER);
-        Py_DECREF(obj);
-        obj = ret;
+        // Testing if this code can be ever hit by existing tests
+        PyErr_SetString(PyExc_RuntimeError, "Not Implemented");
+        // PyObject *ret;
```

Comment:
Coverage agrees: this error is never hit. Is there a reason to leave the commented out code (other than the comment on line 1875)?

Reply:
This is just rough code change. Now we know that this is never gonna hit, let me know what should I do here for never copy stuff. Should I remove checking of never copy flag here and keep the rest of the code block as it is?

Reply:
I would prefer to remove all the dead code, leaving only the runtime error check.

Reply:
Okay. Sure. The error message seems right? Let me know if anything else is to be done. I will remove everything except, `PyErr_SetString(PyExc_RuntimeError, "Not Implemented");` in this `if` block.